src/dmd/mars.d

```diff
@@ -836,13 +836,44 @@ extern (C++) void generateJson(Modules* modules)
     }
 }
 
+version (DigitalMars)
+{
+    private void installMemErrHandler()
+    {
+        // (only available on some platforms on DMD)
+        const shouldDoMemoryError = getenv("DMD_INSTALL_MEMERR_HANDLER");
+        if (shouldDoMemoryError !is null && *shouldDoMemoryError == '1')
+        {
+            import etc.linux.memoryerror;
+            static if (is(typeof(registerMemoryErrorHandler())))
+            {
+                registerMemoryErrorHandler();
+            }
+            else
+            {
+                printf("**WARNING** Memory error handler not supported on this platform!\n");
```

Comment:
I'm not sure if we will be always enable this feature for all CI? I mean, I have definitely seen problems with this in some cases that made the code worse than not having it installed.

In which case, it's useful for using the CI to find issues (like the original #11574), or to use it locally when a segfault happens.

Perhaps I can change the code to configure the printout based on the value of the string? Like if the environment variable is not blank, then enable the memory handler, and if the value is not "NOPRINT", then print out the message?

Reply:
It doesn't need to be enabled for all CI's. I thought we could e.g. enable it e.g. for the `-debug` builds done on Circle and Azure* (unless it causes errors, etc).

> Perhaps I can change the code to configure the printout based on the value of the string? Like if the environment variable is not blank, then enable the memory handler, and if the value is not "NOPRINT", then print out the message?

Something like that, allthough it could simply use `1` for the current behaviour and `2` for silent mode (I doubt that the silent mode is ever relevant outside of our CI's)